Design fix

ayomide321 opened the pull request:
-Fixed the web page design to include logging in and logging out
-Replaced filler text with an actual objective

**ISSUE RESOLVED**

Commit:
Removed excess signin/signout buttons, and replaced filler text with objective from docs.

Commit:
Fixed CSS Bug

flying-bus commented on maptivist/src/main/webapp/script.js:
```diff
@@ -95,6 +100,17 @@ var GoogleAuth;
     }
   }
 
+  function handleAuthClick2() {
```
Can you add documentation for the the difference between handleAuthClick and handleAuthClick2?

flying-bus commented on maptivist/src/main/webapp/script.js:
```diff
@@ -79,8 +79,13 @@ var GoogleAuth;
-      $('#revoke-access-button').click(function() {
-        revokeAccess();
+
+      $('#myButton').click(function() {
```
If I understand the code correctly, at first, both buttons says "Sign In" (or something similar). And then after the user signs in:

setSignInStatus() changes the text of the top "Sign In" button to  "Sign out" and the lower one becomes "Become a maptivist".

Then afterwards in this section, the button callbacks are changed to match ("Sign out" gets the sign out function and the "become a maptivist" get the open form function?)

I think the code would be clearer if it were organized such that the button callbacks and names were changed in the same place.

ayomide321 replied:
You got the functionality right, I realized now that I was so focused on getting it to work I forgot to actually make it readable for other people. Ill work on that now. As for pulling from master, I use git pull
.

ayomide321 replied:
I managed to fix the issue by using GitHub Desktop on my personal PC and merging it through there, then pulling on my work laptop afterwards. This is unusual as I am always able to use "git pull".

flying-bus commented:
Ayo, for the branch merging - are you doing a `git pull` or `git fetch; git merge`?

ayomide321 committed:
Merge branch 'master' into designFix

Commit:
Fixed the merge, added documentation on functions, rearranged code to be more readable

flying-bus commented on maptivist/src/main/webapp/index.html:
```diff
@@ -25,18 +25,21 @@
 
     <div class = "header">
-		<button id="sign-in-or-out-button" style="margin-left: 25px">Sign In/Authorize</button>
+		<!--<button id="sign-in-or-out-button" style="margin-left: 25px">Sign In/Authorize</button>-->
```
Delete this instead of leaving commented out the code, please.

flying-bus commented on maptivist/src/main/webapp/script.js:
```diff
@@ -248,26 +244,47 @@ var GoogleAuth;
   function setSigninStatus(isSignedIn) {
     var user = GoogleAuth.currentUser.get();
     var isAuthorized = user.hasGrantedScopes(SCOPE);
+    $('#sign-in-or-out-button').click(function() {
+        handleAuthClick();
+      });
```
Hmm, isn't this button gone now? Can we just delete it?

flying-bus commented on maptivist/src/main/webapp/script.js:
```diff
@@ -248,26 +244,47 @@ var GoogleAuth;
+    $('#sign-in-or-out-button').click(function() {
+        handleAuthClick();
+      });
+      $('#myButton').click(function() {
+        handleAuthClick2();
+      });
+      $('#authLink').click(function() {
+        handleAuthClick();
+      });
```
Now that I re-read the code, I think there might be even a simpler way to do this, and help it be readable.

More informative function names will help the reader understand their purpose.
handleAuthClick -> signInOrOut
handleAuthClick2 ->signInOrOpenMap

Also, I overlooked this before (sorry) and perhaps I was confused by the control flow, but the callbacks don't actually change. They just check the isSignedIn status. In that case, they never need to change. In that case, maybe they can just be put back where they were, or perhaps just put them in the HTML as `onclick` handlers?

Commit:
setting up flagging function

Commit:
Master was updated

Merge branch 'master' of https://github.com/googleinterns/step125-2020 into designFix

TBarrett553 commented on maptivist/src/main/webapp/index.html:
```diff
@@ -25,18 +24,21 @@
+            The problem with becoming an activist is that it is dangerous, because of this less people can get involved with the movement. In addition to the danger, the fact that we are in the midst of a pandemic also factors into why most people can’t go out and protest. Maptivist aims to remove any problems like this, and become sort of a “middle man” in terms of activism. In the age where eve ...
+            <br>
+            <br>
+            Especially in the current state of the United States, it is clear that people are looking to engage in organized activism. People have and continue to protest in the streets to make their voices heard. However, there are many groups of people who are not able to physically participate in local movements. Our project, Maptivist, aims to virtually connect our users to the physical place ...
```
It's my error from the design doc but this line should be corrected:
"The web application will allow verified users to create and place markers on a Google Map to represent the location[s] of protest[s], meetings or important events about local issues."

TBarrett553 reviewed (approved):
lgtm! Just check out my typo

flying-bus commented on maptivist/src/main/webapp/script.js:
```diff
@@ -176,17 +176,33 @@ function createInfowindow(position) {
+	var flagWindow = new google.maps.InfoWindow({
+    content: flagString
+  });
+
```
I think I figured out the issue after doing some debugging. The flag-window is being draw, it's just appearing off the map. I think there are two ways to fix it.

1) Figure out how to pass the marker into this function, and then call:
`figureWindow.open(map, marker)`

Or 2) Figure out how to pass the coordinates of the marker into this function, and then call:

```
flagWindow.setPosition(coordinates);
flagWindow.open(map);
```

Or 3) a quick and dirty thing would be:
```
  flagWindow.setPosition(map.getCenter());
  flagWindow.open(map);
```
But then the info window isn't quite in the right place.

Commit:
Merge branch 'master' of https://github.com/googleinterns/step125-2020 into designFix

Commit:
Added flag form popup

flying-bus commented on maptivist/src/main/webapp/script.js:
```diff
@@ -176,17 +176,39 @@ function createInfowindow(position) {
     </div>
     <div class="flag">
-        <button>Flag</button>
+        <button onclick = "flagFunction()">Flag</button>
```
So, I would pass the marker into `createInfoWindow(position, marker)`

Then for the button, add an `id="info_window_flag_button"`:

And then you can add a `domready` listener to the infowindow:
https://developers.google.com/maps/documentation/javascript/reference/3.40/info-window?hl=en#InfoWindow.domready

Finally, add a `domready` listener
```
infowindow.domready = function() {
   var flag_btn =   document.getElementById("info_window_flag_button");
   flag_btn.onclick = function() {
      // the closure captures the marker.
      flagFunction(marker);
   }
}
```

I *think* something like that should work.

ayomide321 replied:
I think it does work, but until markers can be successfully created I won't know for sure. I've committed and pushed the changes though and will return to this PR when markers can be successfully implemented
